packages/advanced-logic/src/extensions/payment-network/any-to-erc20-proxy.ts

```diff
@@ -48,397 +28,163 @@ const supportedCurrencies: Record<string,  Record<RequestLogicTypes.CURRENCY, st
-    !isValidAddress(addRefundAddressParameters.refundAddress)
-  ) {
-    throw Error('refundAddress is not a valid ethereum address');
-  }
-
-  return ReferenceBased.createAddRefundAddressAction(
-    ExtensionTypes.ID.PAYMENT_NETWORK_ANY_TO_ERC20_PROXY,
-    addRefundAddressParameters,
-  );
-}
-
-/**
- * Creates the extensionsData to add a fee address
- *
- * @param addFeeParameters extensions parameters to create
- *
- * @returns IAction the extensionsData to be stored in the request
- */
-function createAddFeeAction(
-  addFeeParameters: ExtensionTypes.PnFeeReferenceBased.IAddFeeParameters,
-): ExtensionTypes.IAction {
-  if (addFeeParameters.feeAddress && !isValidAddress(addFeeParameters.feeAddress)) {
-    throw Error('feeAddress is not a valid ethereum address');
-  }
-
-  if (addFeeParameters.feeAmount && !Utils.amount.isValid(addFeeParameters.feeAmount)) {
-    throw Error('feeAmount is not a valid amount');
-  }
-
-  if (!addFeeParameters.feeAmount && addFeeParameters.feeAddress) {
-    throw Error('feeAmount requires feeAddress');
-  }
-  if (addFeeParameters.feeAmount && !addFeeParameters.feeAddress) {
-    throw Error('feeAddress requires feeAmount');
-  }
-
-  return {
-    action: ExtensionTypes.PnFeeReferenceBased.ACTION.ADD_FEE,
-    id: ExtensionTypes.ID.PAYMENT_NETWORK_ANY_TO_ERC20_PROXY,
-    parameters: addFeeParameters,
-  };
-}
-/**
- * Applies the extension action to the request state
- * Is called to interpret the extensions data when applying the transaction
- *
- * @param extensionsState previous state of the extensions
- * @param extensionAction action to apply
- * @param requestState request state read-only
- * @param actionSigner identity of the signer
- *
- * @returns state of the request updated
- */
-function applyActionToExtension(
-  extensionsState: RequestLogicTypes.IExtensionStates,
-  extensionAction: ExtensionTypes.IAction,
-  requestState: RequestLogicTypes.IRequest,
-  actionSigner: IdentityTypes.IIdentity,
-  timestamp: number,
-): RequestLogicTypes.IExtensionStates {
-  checkSupportedCurrency(requestState.currency, extensionAction.parameters.network || 'rinkeby');
-
-  const copiedExtensionState: RequestLogicTypes.IExtensionStates = Utils.deepCopy(extensionsState);
-
-  if (extensionAction.action === ExtensionTypes.PnFeeReferenceBased.ACTION.CREATE) {
-    if (requestState.extensions[extensionAction.id]) {
-      throw Error(`This extension has already been created`);
+export class anyToErc20ProxyPaymentNetwork extends erc20FeeProxyPaymentNetwork {
+  public constructor() {
+    super();
+    this.currentVersion = CURRENT_VERSION;
+    this.paymentNetworkId = ExtensionTypes.ID.PAYMENT_NETWORK_ANY_TO_ERC20_PROXY;
+  }
+
+  /**
+   * Creates the extensionsData to create the extension Any to ERC20
+   *
+   * @param creationParameters extensions parameters to create
+   *
+   * @returns IExtensionCreationAction the extensionsData to be stored in the request
+   */
```

Comment:
I'm not sure the comments are useful on the child classes. Opinions welcome!

Reply:
I think child methods should have comments. Better for IDE help at least